[quant][graphmode][refactor] Factor out common parts of general value ops

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #38359 f[quant][graphmode][refactor] Factor out getFixedQParamOpFusionInfo
* #38358 [quant][graphmode][refactor] Factor out getInputTensorQParamOpFusionInfo
* #38278 [quant][graphmode] Support ops with fixed quantization parameters
* #38277 [quant][graphmode][refactor] insertDeQuantForAllUse
* #38276 [quant][graphmode][refactor] Refactor propagateQuantizationOps
* #38166 [quant][graphmode] Move leaky_relu to general value op map
* #38165 [quant][graphmode] Move elu to general value ops map
* #38164 [quant][graphmode] Move hardtanh to general value ops map
* #38163 [quant][graphmode] Move clamp to general value ops map
* #38162 [quant][graphmode] Move interpolate to general value ops
* **#38161 [quant][graphmode][refactor] Factor out common parts of general value ops**
* #38160 [quant][graphmode] Move `aten::mean` to general value ops
* #38330 [quant][graphmode] Move avg_pool/adaptive_avg_pool to general value ops

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21512972](https://our.internmc.facebook.com/intern/diff/D21512972)

jerryzh168 committed:
[quant][graphmode][refactor] Factor out common parts of general value ops

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out common parts of general value ops"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar commented on torch/csrc/jit/passes/quantization/quantization_patterns.h:
```diff
@@ -440,17 +440,21 @@ graph(%a_quant, %normalized_shape, %weight, %bias, %eps, %cudnn_enabled, %output
-          %a_dequant = aten::dequantize(%a_quant)
-          %r = aten::avg_pool1d(%a_dequant, %kernel_size, %stride, %padding, %ceil_mode, %count_include_pad, %divisor_override)
+  // IR pattern common to all general value ops
+  std::string common_general_value_op = R"(
```
Maybe call it common_suffix to be more specific?

jerryzh168 replied:
OK

jerryzh168 replied:
will have a larger refactor later that factors out a function, will rename this at that time.

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out common parts of general value ops"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21512972](https://our.internmc.facebook.com/intern/diff/D21512972)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out common parts of general value ops"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21512972](https://our.internmc.facebook.com/intern/diff/D21512972)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out common parts of general value ops"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21512972](https://our.internmc.facebook.com/intern/diff/D21512972)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out common parts of general value ops"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21512972](https://our.internmc.facebook.com/intern/diff/D21512972)

[ghstack-poisoned]
